test/addons_test.go

```diff
@@ -0,0 +1,49 @@
+package test
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/blang/semver"
+
+	"github.com/mesosphere/kubeaddons/api/v1beta1"
+	"github.com/mesosphere/kubeaddons/hack/temp"
+	"github.com/mesosphere/kubeaddons/pkg/test"
+	"github.com/mesosphere/kubeaddons/pkg/test/cluster/kind"
+)
+
+// TestAddons tests deployment of all addons in this repository
+func TestAddons(t *testing.T) {
+	cluster, err := kind.NewCluster(semver.MustParse("1.16.3"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cluster.Cleanup()
+
+	if err := temp.DeployController(cluster); err != nil {
+		t.Fatal(err)
+	}
+
+	b, err := ioutil.ReadFile("../addons/zookeeper/0.x/zookeeper.yaml")
```

Comment:
As per the description, this is just step one, another PR will change coverage to ALL addons but there's some Catalog functionality that is not ready yet for this.